src/bridge/cockpitpeer.c

```diff
@@ -95,15 +96,50 @@ G_DEFINE_TYPE (CockpitPeer, cockpit_peer, G_TYPE_OBJECT);
 
 static void
 startup_done (CockpitPeer *self,
-              const gchar *problem)
+              const gchar *problem,
+              const gchar *stderr)
 {
   if (self->startup_done_function)
     {
-      self->startup_done_function (problem, self->startup_done_data);
+      self->startup_done_function (problem, stderr, self->startup_done_data);
       self->startup_done_function = NULL;
     }
 }
 
+static gchar *
+startup_get_stderr_and_reset (CockpitPeer *self)
+{
+  CockpitPipe *pipe;
+  GByteArray *buffer;
+  GBytes *bytes;
+  GBytes *clean;
+  gchar *data;
+  gsize length;
+
+  pipe = cockpit_pipe_transport_get_pipe (COCKPIT_PIPE_TRANSPORT (self->other));
+
+  buffer = cockpit_pipe_get_stderr (pipe);
```

Comment:
This is a copy/paste, let's factor it out into a `cockpit_pipe_take_stderr_as_utf8` function.